### PR TITLE
Swap to always loading the configuration module shipped in the PS Version task

### LINKED ISSUE #1042: Version PowerShell task fails if naming clash exists with Configuration module
### Azure DevOps Extensions

Manifest Versioning

### Platform

Azure DevOps Services

### Azure DevOps Server (TFS) Version

_No response_

### Extension Version

_No response_

### Describe the bug

```bash
The Version Powershell task relies on a module named Configuration. If another module exists on the machine which defines some of the same commands, execution of the task fails as Install-Module does not allow modules to be installed with name clashes unless -AllowClobber is specified.
```

### Repo Steps

```bash
1. Install PSDepend on a build server from the PowerShell gallery
2. Attempt to run a build which versions a PowerShell file using the task
```

### Expected Behavior

```bash
Either: 
- The task should run in this situation.
Or:
- There should be a flag I can set on the task to allow me to override the default behaviour of Install-Module when it encounters a naming clash.
```

### Logging Information

```bash
Excerpt of Azure Pipelines build logs:

##[debug]00:00:10.4205656 PowerShell Script 'PSModule' Function 'Install-Package' returns null.
##[debug]Caught exception from task script.
##[debug]Error record:
##[debug]PackageManagement\Install-Package : The following commands are already available on this system:'Add-MetadataConverter,ConvertFrom-Metadata,ConvertTo-Metadata,Export-Metadata,Get-ManifestValue,Import-Metadata,Update-Manifest'. This module 'Metadata' may override the existing commands. If you still want to install this module 'Metadata', use -AllowClobber parameter.
##[debug]At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\2.2.5\PSModule.psm1:9711 char:34
##[debug]+ ... talledPackages = PackageManagement\Install-Package @PSBoundParameters
##[debug]+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
##[debug]    + CategoryInfo          : InvalidOperation: (Microsoft.Power....InstallPackage:InstallPackage) [Install-Package],     Exception
##[debug]    + FullyQualifiedErrorId : CommandAlreadyAvailable,Validate-ModuleCommandAlreadyAvailable,Microsoft.PowerShell.Pack    ageManagement.Cmdlets.InstallPackage
##[debug] 
##[debug]Script stack trace:
##[debug]at Install-Module<Process>, C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\2.2.5\PSModule.psm1: line 9711
##[debug]at <ScriptBlock>, E:\Agent\_work\_tasks\VersionPowerShellModule_036278fe-8a47-426b-b5b3-bef05334db00\2.29.16\Update-PowerShellModuleVersion.ps1: line 89
##[debug]at <ScriptBlock>, <No file>: line 1
##[debug]at <ScriptBlock>, <No file>: line 22
##[debug]at <ScriptBlock>, <No file>: line 18
##[debug]at <ScriptBlock>, <No file>: line 1
##[debug]Exception:
##[debug]System.Exception: The following commands are already available on this system:'Add-MetadataConverter,ConvertFrom-Metadata,ConvertTo-Metadata,Export-Metadata,Get-ManifestValue,Import-Metadata,Update-Manifest'. This module 'Metadata' may override the existing commands. If you still want to install this module 'Metadata', use -AllowClobber parameter.
##[error]The following commands are already available on this system:'Add-MetadataConverter,ConvertFrom-Metadata,ConvertTo-Metadata,Export-Metadata,Get-ManifestValue,Import-Metadata,Update-Manifest'. This module 'Metadata' may override the existing commands. If you still want to install this module 'Metadata', use -AllowClobber parameter.
```

### PR DESCRIPTION
**Pull request by rfennell**
- Removes the dependency on PSGallery for the VerisonPoweshellTask - fixes #1042
- Updates all NPM packages (to address vulnerabilities identified in SYNK) on all Node based tasks in the extension - required else the pipeline fails

### CODE REVIEW

File: Extensions/Versioning/VersionPowerShellModule/src/Update-PowerShellModuleVersion.ps1 [hunk 1]
```diff
@@ -65,37 +65,9 @@ if ($Prerelease) {
...
-        $Null = Get-Command -Module Configuration
-    }
-}
-else {
-    Write-Verbose -Message "PowerShellGet is unavailable, using Configuration module shipped with task instead"
-    Import-Module "$PSScriptRoot\1.3.0\Configuration.psd1" -force
-    $Null = Get-Command -Module Configuration
-}
+Write-Verbose -Message "Loading Configuration module shipped with tasks"
+Import-Module "$PSScriptRoot\1.3.0\Configuration.psd1" -force
```
Comment (ChrisLGardner) [inline Extensions/Versioning/VersionPowerShellModule/src/Update-PowerShellModuleVersion.ps1:69]: This should be 1.5.0 not 1.3.0

Comment (ChrisLGardner) [reply Extensions/Versioning/VersionPowerShellModule/src/Update-PowerShellModuleVersion.ps1:69]: Might be better to nest it in a folder called Configuration in the same path then you can just `Import-Module $PSScriptRoot\Configuration` and it'll work for whatever version gets placed there.

Comment (rfennell) [reply Extensions/Versioning/VersionPowerShellModule/src/Update-PowerShellModuleVersion.ps1]: Ops cut and paste error, my bad

**Comment by rfennell** [comment]
Well, this is proving troublesome. I have swapped the code so that the 1.5.0 configuration module shipped in the task is always used. This has a dependency on the metadata module. 

If I run the script locally it works in all the PowerShell versions I have on my PC. It also work on a locally hosted build agent, but if I run it via a MS Hosted agent it fails

```
##[debug]Loading module from path 'D:\a\_tasks\VersionPowerShellModule_036278fe-8a47-426b-b5b3-bef05334db00\2.30.10\Configuration\1.5.0\Configuration.psd1'.
##[debug]Caught exception from task script.
##[debug]Error record:
##[debug]Import-Module : The required module 'Metadata' is not loaded. Load the module or remove the module from 'RequiredModules' in the file 'D:\a\_tasks\VersionPowerShellModule_036278fe-8a47-426b-b5b3-bef05334db00\2.30.10\Configuration\1.5.0\Configuration.psd1'.
##[debug]At D:\a\_tasks\VersionPowerShellModule_036278fe-8a47-426b-b5b3-bef05334db00\2.30.10\Update-PowerShellModuleVersion.ps1:69 char:1
```
I tried 
- putting the configuration and metadata modules  in their own folders and loading them both explicitly
- I also tried putting them all in one folder

Neither work.

Any ideas @ChrisLGardner.

Just to make it seem a little stranger. On my locally hosted build agent the Pester 10.3.10 task is also failing with the error

> VERBOSE: Importing function 'Import-Pester'.
>
> ##[error]Get-Command : The term 'Install-Module' is not recognized as the name of a cmdlet, function, script file, or operable 
> program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
>
> ##[debug]Processed: ##vso[task.issue type=error;]Get-Command : The term 'Install-Module' is not recognized as the name of a cmdlet, function, script file, or operable %0D%0Aprogram. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.%0D%0A
> ##[debug]task result: Failed
> 

Hope there is not an issue there too

**Comment by ChrisLGardner** [comment]
An easy fix there would be to store a copy of the Metadata module in the task folder and then update the PSModulePath within the task to include this folder and it should resolve that, you can do that by adding `$Env:PSModulePath = "$PSScriptRoot;$Env:PSModulePath"` before the Import-Module call. 

I think this module used to be a nested module or something within the Configuration module which would explain why we haven't seen this before.

I'll have a look at the Pester issue, I don't think I've seen anyone else reporting problems with it but I'll run some test pipelines.

**Comment by rfennell** [comment]
Sorting the PSModulePath has fixed the issue on the hosted agents. The revised task also work on my private agent.

As to the Pester issue, it seems to be local to my machine, s I would not worry about it too much

Thanks for the help. I will get this fix shipped

**Pull request merged**